addons/mass_mailing_sms/tests/test_mailing_retry.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.mass_mailing_sms.tests.common import MassSMSCommon
+from odoo.addons.base.tests.test_ir_cron import CronMixinCase
+from odoo.tests.common import users
+
+class TestMailingRetrySMS(MassSMSCommon, CronMixinCase):
+
+    @users('user_marketing')
+    def test_sms_retry_immediate_trigger(self):
+        default_list_ids = self.env['mailing.list'].with_context(self._test_context).create({
+            'name': 'ListC',
+            'contact_ids': [
+                (0, 0, {'name': 'test', 'email': 'test@odoo.com', 'mobile': '+91 1234567890'}),
+            ]
+        }).ids
+        mailing = self.env['mailing.mailing'].create({
+            'name': 'TestMailing',
+            'subject': 'Test',
+            'mailing_type': 'sms',
+            'body_plaintext': 'Coucou hibou',
+            'mailing_model_id': self.env['ir.model']._get('res.partner').id,
+            'contact_list_ids': default_list_ids,
+        })
+        mailing.action_send_sms()
+
+        # Send the SMS, which leads to RETRY because IAP server is not available
+        self.env.ref('sms.ir_cron_sms_scheduler_action').sudo().method_direct_trigger()
```

Comment:
Same logic here.
It's especially important since we want to make sure the test does not contact our IAP server (would create unnecessary spam / overhead).

The same kind of logic as proposed in mass_mailing could be applied.